### PR TITLE
Resolve abbreviations in Ltac2 eval reduction flags

### LINKED ISSUE #19590: Abbreviations in Ltac2 eval reduction flags
### Description of the problem

The following fails (fixed by https://github.com/coq/coq/pull/19589 )

### Small Coq file to reproduce the bug

```coq
From Ltac2 Require Import Ltac2.

Definition a := 3.
Notation b := a.

Goal a = 3.
let ea := eval cbv delta [b] in a in change a with $ea.
(* Error: The reference b was not found in the current environment. *)
```

### Version of Coq where this bug occurs

8.20.0

### Last version of Coq where the bug did not occur

_No response_

### PR DESCRIPTION
**Pull request by proux01**
Noticed in https://github.com/mit-plv/rewriter/pull/159

Fixes / closes #19590

- [x] Added / updated **test-suite**.

- [x] Added **changelog**.
- [ ] ~Added / updated **documentation**.~
  
  - [ ] ~Documented any new / changed **user messages**.~
  - [ ] ~Updated **documented syntax** by running `make doc_gram_rsts`.~

- [ ] ~Opened **overlay** pull requests.~

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

**Review by JasonGross — APPROVED** [review]
Plausible the changelog should have "Fixed" rather than "Added", but LGTM either way

File: doc/changelog/06-Ltac2-language/19589-ltac2-eval-flags.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- **Fixed:**
+  resolution of :ref:`abbreviations <Abbreviations>` in Ltac2 eval tactic
+  delta-reduction flags :token:`ltac2_delta_reductions`
```
Comment (SkySkimmer) [inline doc/changelog/06-Ltac2-language/19589-ltac2-eval-flags.rst:3]: I believe from the code that it's not just the reduction flags, it's anything that parses qualid to produce a globref

Comment (proux01) [reply doc/changelog/06-Ltac2-language/19589-ltac2-eval-flags.rst:4]: Indeed, fixed

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**